tools/system_libs.py

```diff
@@ -1333,22 +1333,25 @@ def can_use(self):
 
 
 class libwasm_workers(MTLibrary):
+  name = 'libwasm_workers'
+
   def __init__(self, **kwargs):
     self.debug = kwargs.pop('debug')
     super().__init__(**kwargs)
 
-  name = 'libwasm_workers'
-
   def get_cflags(self):
-    cflags = get_base_cflags() + ['-D_DEBUG' if self.debug else '-Oz']
+    cflags = super().get_cflags()
     if self.debug:
+      cflags += ['-D_DEBUG']
       # library_wasm_worker.c contains an assert that a nonnull paramater
       # is no NULL, which llvm now warns is redundant/tautological.
       cflags += ['-Wno-tautological-pointer-compare']
+      # Override the `-O2` default.  Building library_wasm_worker.c with
+      # `-O1` or `-O2` currently causes tests to fail.
+      # https://github.com/emscripten-core/emscripten/issues/19331
+      cflags += ['-O0']
     else:
-      cflags += ['-DNDEBUG']
-    if self.is_ww or self.is_mt:
-      cflags += ['-pthread', '-sWASM_WORKERS']
```

Comment:
Do we not need `-pthread` and `-sWASM_WORKERS`? (I could imagine upper code sets the first, but surely not the second?)

Reply:
The MTLibrary superclass takes care of that.  I confirmed that both these flags are present both before and after this change.

Reply:
Interesting. I'm surprised it sets `-sWASM_WORKERS` since this is the only wasm-workers library... but I guess it's simpler to always do it.

Reply:
Actually all `MTLibrary` subclass come in `-mt` and `-ww` flavors, not just this one.